src/rust/ensogl/lib/select/src/component.rs

```diff
@@ -0,0 +1,403 @@
+    /// Move selection one position down.
+    move_selection_down,
+    /// Move selection page down (jump over all visible entries).
+    move_selection_page_down,
+    /// Move selection to the last argument.
+    move_selection_to_last,
+    /// Chose the currently selected entry.
+    chose_selected_entry,
+    /// Deselect all entries.
+    deselect_entries,
+}
+
+impl application::command::CommandApi for Select {
+    fn command_api_docs() -> Vec<application::command::EndpointDocs> {
+        Commands::command_api_docs()
+    }
+
+    fn command_api(&self) -> Vec<application::command::CommandEndpoint> {
+        self.frp.input.command.command_api()
+    }
+}
+
+ensogl_text::define_endpoints! {
+    Commands { Commands }
+    Input {
+        resize           (Vector2<f32>),
+        scroll_jump      (f32),
+        set_entries      (entry::AnyModelProvider),
+        select_entry     (entry::Id),
+        chose_entry      (entry::Id),
+        deselect_entries (),
+    }
+    Output {
+        selected_entry  (Option<entry::Id>),
+        chosen_entry    (Option<entry::Id>),
+        size            (Vector2<f32>),
+        scroll_position (f32),
+    }
+}
+
+
+
+// ========================
+// === Select Component ===
+// ========================
+
+/// Select Component.
+///
+/// Select is a displayed list of entries with possibility of selecting one and "chosing" by
+/// clicking or pressing enter.
+#[allow(missing_docs)]
+#[derive(Clone,CloneRef,Debug)]
+pub struct Select {
+    model   : Model,
+    pub frp : Frp,
+}
+
+impl Deref for Select {
+    type Target = Frp;
+    fn deref(&self) -> &Self::Target { &self.frp }
+}
+
+impl Select {
+    /// Constructor.
+    pub fn new(app:&Application) -> Self {
+        let frp           = Frp::new_network();
+        let model         = Model::new(app);
+        Select{frp,model}.init(app)
+    }
+
+    fn init(self, app:&Application) -> Self {
+        const MAX_SCROLL:f32 = entry::HEIGHT/2.0;
+
+        let frp              = &self.frp;
+        let network          = &frp.network;
+        let model            = &self.model;
+        let scene            = app.display.scene();
+        let mouse            = &scene.mouse.frp;
+        let window_y         = Animation::<f32>::new(&network);
+        let selection_y      = Animation::<f32>::new(&network);
+        let selection_height = Animation::<f32>::new(&network);
+
+        frp::extend!{ network
+
+            // === Mouse Position ===
+
+            mouse_in <- mouse.position.all_with(&frp.size, f!([model,scene](pos,size) {
+                let pos_obj_space = scene.screen_to_object_space(&model.background,*pos);
+                let x_range = (-size.x / 2.0)..=(size.x / 2.0);
+                let y_range = (-size.y / 2.0)..=(size.y / 2.0);
+                x_range.contains(&pos_obj_space.x) && y_range.contains(&pos_obj_space.y)
+            }));
+            mouse_moved       <- mouse.distance.map(|dist| *dist > std::f32::EPSILON);
```

Comment:
add here some treshold as a constant. I think that it could not be triggered by 1 px movement.

Reply:
Actually I tested it with threshold of one pixel and it feels bad.